test/unit/renderer/viz/expressions/binary.test.js

```diff
@@ -75,134 +76,496 @@ describe('src/renderer/viz/expressions/binary', () => {
         });
     });
 
-    describe('eval', () => {
-        describe('and', () => {
-            test('and', s.TRUE, s.TRUE, 1);
-            test('and', s.TRUE, s.FALSE, 0);
-            test('and', s.FALSE, s.FALSE, 0);
-            test('and', 0.5, s.TRUE, 0.5);
-            test('and', 0.5, 0.5, 0.25);
+    describe('.value', () => {
+        function testValue (fn, param1, param2, expected) {
+            it(`${fn}(${param1}, ${param2}) should return ${expected}`, () => {
+                const actual = s[fn](param1, param2).value;
+                expect(actual).toEqual(expected);
+            });
+        }
+
+        describe('mul', () => {
+            describe('NUMBERS_TO_NUMBER', () => {
+                testValue('mul', 0, 0, 0);
+                testValue('mul', 1, 0, 0);
+                testValue('mul', 1, 1, 1);
+                testValue('mul', 1, 2, 2);
+                testValue('mul', -1, 2, -2);
+            });
+
+            describe('NUMBER_AND_COLOR_TO_COLOR', () => {
+                testValue('mul', 10, s.rgba(255, 15, 12, 1), { r: 10, g: 1, b: 0, a: 1 });
```

Comment:
When does this NUMBER * COLOR is useful? do you know any example?

Reply:
Actually, no, but it's been there for a while and I _think_ there was an example. I'll do a bit of research.